File: cadwiki-nuget/cadwiki.NUnitTestRunner/UI/FormTestRunner.vb
Imports System.Drawing
Imports System.Windows.Media.Imaging
Imports cadwiki.NetUtils
Imports cadwiki.NUnitTestRunner.Results
Imports cadwiki.NUnitTestRunner.TestEvidence
Imports Application = System.Windows.Forms.Application

Namespace UI
    Public Class FormTestRunner
        Public WithEvents ObservableResults As New ObservableTestSuiteResults

        Private _commonUiObject As New CommonUiObject()

        Private Sub TestMessages_OnChanged(sender As Object, e As EventArgs) Handles ObservableResults.MessageAdded
            Dim suiteResults As Results.ObservableTestSuiteResults = CType(sender, ObservableTestSuiteResults)
            Dim messages As List(Of String) = suiteResults.Messages
            Dim lastItem As String = messages(messages.Count - 1)
            RichTextBoxConsole.AppendText(lastItem)
            Application.DoEvents()
        End Sub


        Private Sub TestResults_OnChanged(sender As Object, e As EventArgs) Handles ObservableResults.ResultAdded
            Dim suiteResults As ObservableTestSuiteResults = CType(sender, ObservableTestSuiteResults)
            Dim testResults As List(Of TestResult) = suiteResults.TestResults
            Dim mostRecentlyAddedTestResult As TestResult = testResults(testResults.Count - 1)
            _commonUiObject.WinFormsAddTreeViewItemForTestResult(mostRecentlyAddedTestResult, TreeViewResults)
        End Sub



        Public Sub New()
            InitializeComponent()
            Init()
        End Sub

        Public Sub Init()
            RichTextBoxConsole.AppendText(vbLf & "NunitTestRunner started")
            Dim bitMap As Bitmap = FileStore.My.Resources.ResourceIcons._500x500_cadwiki_v1
            Dim icon As Icon = Bitmaps.BitmapToIcon(bitMap, True, Color.White)
            Me.Icon = icon
        End Sub

        Public Sub New(ByRef suiteResults As ObservableTestSuiteResults)
            ObservableResults = suiteResults
            InitializeComponent()
            Init()
        End Sub

        Private Sub ButtonOk_Click(sender As Object, e As EventArgs) Handles ButtonOk.Click
            Me.Close()
        End Sub

        Private Sub Cancel_Click(sender As Object, e As EventArgs) Handles Cancel.Click
            Me.Close()
        End Sub

        Private Sub ButtonOpenTestEvidenceFolder_Click(sender As Object, e As EventArgs) Handles ButtonOpenTestEvidenceFolder.Click
            Dim creator As New Creators.TestEvidenceCreator()
            Dim foldername As String = creator.GetFolderCache()
            Process.Start(foldername)
        End Sub
    End Class

End Namespace

File: cadwiki-nuget/cadwiki.NUnitTestRunner/UI/FormTestRunner.cs
using System;
using System.Diagnostics;
using System.Drawing;
using System.Runtime.CompilerServices;
using Application = System.Windows.Forms.Application;
using cadwiki.NetUtils;
using cadwiki.NUnitTestRunner.Results;
using Microsoft.VisualBasic;

namespace cadwiki.NUnitTestRunner.UI
{
    public partial class FormTestRunner
    {
        private ObservableTestSuiteResults _ObservableResults;

        public virtual ObservableTestSuiteResults ObservableResults
        {
            [MethodImpl(MethodImplOptions.Synchronized)]
            get
            {
                return _ObservableResults;
            }

            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                if (_ObservableResults != null)
                {
                    _ObservableResults.MessageAdded -= TestMessages_OnChanged;
                    _ObservableResults.ResultAdded -= TestResults_OnChanged;
                }

                _ObservableResults = value;
                if (_ObservableResults != null)
                {
                    _ObservableResults.MessageAdded += TestMessages_OnChanged;
                    _ObservableResults.ResultAdded += TestResults_OnChanged;
                }
            }
        }

        private CommonUiObject _commonUiObject = new CommonUiObject();

        private void TestMessages_OnChanged(object sender, EventArgs e)
        {
            ObservableTestSuiteResults suiteResults = (ObservableTestSuiteResults)sender;
            var messages = suiteResults.Messages;
            string lastItem = messages[messages.Count - 1];
            RichTextBoxConsole.AppendText(lastItem);
            Application.DoEvents();
        }


        private void TestResults_OnChanged(object sender, EventArgs e)
        {
            ObservableTestSuiteResults suiteResults = (ObservableTestSuiteResults)sender;
            var testResults = suiteResults.TestResults;
            var mostRecentlyAddedTestResult = testResults[testResults.Count - 1];
            _commonUiObject.WinFormsAddTreeViewItemForTestResult(mostRecentlyAddedTestResult, TreeViewResults);
        }



        public FormTestRunner()
        {
            ObservableResults = new ObservableTestSuiteResults();
            InitializeComponent();
            Init();
        }

        public void Init()
        {
            RichTextBoxConsole.AppendText(Environment.NewLine + "NunitTestRunner started");
            Bitmap bitMap = cadwiki.FileStore.ResourceIcons._500x500_cadwiki_v1;
            var icon = Bitmaps.BitmapToIcon(bitMap, true, Color.White);
            Icon = icon;
        }

        public FormTestRunner(ref ObservableTestSuiteResults suiteResults)
        {
            ObservableResults = new ObservableTestSuiteResults();
            ObservableResults = suiteResults;
            InitializeComponent();
            Init();
        }

        private void ButtonOk_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void Cancel_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void ButtonOpenTestEvidenceFolder_Click(object sender, EventArgs e)
        {
            var creator = new Creators.TestEvidenceCreator();
            string foldername = creator.GetFolderCache();
            Process.Start(foldername);
        }
    }

}